### PR TITLE
Message propagation integration test

### PR DESCRIPTION
**Pull request by harbu**
_Note_: Depends on message-numbering branch.

Integration test that runs a scenario (see fix #41) to verify that message propagation works in network. Two things are checked in particular:
- Messages are propagated from one node to another
- StreamIds are obeyed

### CODE REVIEW

**Comment by mirpo** [comment]
travis failed :)

File: test/integration/message-propagation.test.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
+const TrackerNode = require('../../src/protocol/TrackerNode')
+const TrackerServer = require('../../src/protocol/TrackerServer')
+const { startTracker, startNetworkNode } = require('../../src/composition')
+const { callbackToPromise, BOOTNODES } = require('../../src/util')
+const { wait, waitForEvent, LOCALHOST } = require('../../test/util')
+
+jest.setTimeout(90000)
+
+describe('message propagation in network', () => {
```
Comment (mirpo) [inline test/integration/message-propagation.test.js:12]: maybe let's split this test into reusable functions. just idea.

**Review by mirpo — APPROVED** [review]

**Comment by harbu** [comment]
Fixed to work with new master. Also replaced `NetworkNode` with `Node`.

Currently gets stuck in infinite message sending loop. I'm assuming this is because of lack of message numbering.

**Comment by mirpo** [comment]
is it ready for review?

**Comment by harbu** [comment]
Depends on #92 to be fixed.

**Review by ptesavol — APPROVED** [review]
ok

**Review by mirpo — APPROVED** [review]

**Pull request merged**